src/videojs-contrib-hls.js

```diff
@@ -328,6 +330,10 @@ class HlsHandler extends Component {
       this.selectPlaylist ?
         this.selectPlaylist.bind(this) : Hls.STANDARD_PLAYLIST_SELECTOR.bind(this);
 
+    this.masterPlaylistController_.initialSelectPlaylist =
+      this.selectPlaylist ?
```

Comment:
Good catch, I reworked things a bit (I set the selection in the master-playlist-controller) and must have mucked that up.